libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/building/api/MapboxBuildingsApi.kt

```diff
@@ -0,0 +1,119 @@
+import com.mapbox.navigation.ui.maps.building.BuildingAction
+import com.mapbox.navigation.ui.maps.building.BuildingProcessor
+import com.mapbox.navigation.ui.maps.building.model.BuildingError
+import com.mapbox.navigation.ui.maps.building.model.BuildingValue
+import com.mapbox.navigation.utils.internal.JobControl
+import com.mapbox.navigation.utils.internal.ThreadController
+import com.mapbox.navigation.utils.internal.ifNonNull
+import kotlinx.coroutines.launch
+
+/**
+ * Mapbox Buildings Api allows you to query buildings on a [MapboxMap] that you would like
+ * to highlight.
+ * @property mapboxMap
+ */
+class MapboxBuildingsApi internal constructor(
+    private val mapboxMap: MapboxMap,
+    private val processor: BuildingProcessor
+) {
+
+    constructor(mapboxMap: MapboxMap) : this(mapboxMap, BuildingProcessor)
+
+    private val mainJobController: JobControl by lazy { ThreadController.getMainScopeAndRootJob() }
+
+    /**
+     * The API can be invoked to query a building on [MapboxMap] using the [point] provided as
+     * an input. The API returns a [BuildingError] if there is an internal issue fetching the buildings.
+     * Otherwise, it returns buildings wrapped inside [BuildingValue] in a form of list of [QueriedFeature].
+     * This list is empty if the building is not found on the [MapboxMap]
+     *
+     * Note: There are two kind of points that can be used here to query a building.
+     * - A point you can obtain that lives on a building also known as building location coordinate
+     * - A routable point on the road to where you would be routed/navigated to.
+     *
+     * The API will always return an empty list of buildings if you pass in the routable point.
+     */
+    fun queryBuildingToHighlight(
+        point: Point,
+        callback: MapboxNavigationConsumer<Expected<BuildingError, BuildingValue>>
+    ) {
+        val action = BuildingAction.QueryBuilding(point, mapboxMap)
+        mainJobController.scope.launch {
+            val result = processor.queryBuilding(action)
+            callback.accept(result.queriedBuildings)
+        }
+    }
+
+    /**
+     * The API can be invoked to query a building on [MapboxMap] using the [RouteProgress] provided as
+     * an input. Use this function to query a building when you reach a waypoint in a multi-leg trip.
+     * The API returns a [BuildingError] if there is an internal issue fetching the buildings.
+     * Otherwise, it returns buildings wrapped inside [BuildingValue] in a form of list of [QueriedFeature].
+     * This list is empty if the building is not found on the [MapboxMap]
+     *
+     * Note: There are two kind of points that can be used here to query a building.
+     * - A point you can obtain that lives on a building also known as building location coordinate
+     * - A routable point on the road to where you would be routed/navigated to.
+     *
+     * The API will always return an empty list of buildings if you pass in the routable point.
```

Comment:
Let's explain that in other for this method to work, your route's original request should be a "location point" and not a "routing point", otherwise, nothing will be returned.

If the coordinate is a routing point (which is the correct way to do it), you should define a `"waypoint_targets"` in the original request, for each of the waypoints, and for the final destination.